src/materialized/dependencies.rs

```diff
@@ -1720,19 +1720,19 @@ mod test {
                 ",
                 projection: &["year"],
                 expected_plan: vec![
-                    "+--------------+--------------------------------------------------+",
-                    "| plan_type    | plan                                             |",
-                    "+--------------+--------------------------------------------------+",
-                    "| logical_plan | Union                                            |",
-                    "|              |   Projection: coalesce(t1.year, t2.year) AS year |",
-                    "|              |     Full Join: Using t1.year = t2.year           |",
-                    "|              |       SubqueryAlias: t1                          |",
-                    "|              |         Projection: t1.column1 AS year           |",
-                    "|              |           TableScan: t1 projection=[column1]     |",
-                    "|              |       SubqueryAlias: t2                          |",
-                    "|              |         TableScan: t2 projection=[year]          |",
-                    "|              |   TableScan: t3 projection=[year]                |",
-                    "+--------------+--------------------------------------------------+",
+                    "+--------------+--------------------------------------------------------------------+",
+                    "| plan_type    | plan                                                               |",
+                    "+--------------+--------------------------------------------------------------------+",
+                    "| logical_plan | Union                                                              |",
+                    "|              |   Projection: coalesce(CAST(t1.year AS Utf8View), t2.year) AS year |",
```

Comment:
After https://github.com/apache/datafusion/pull/16906, the cast will be unnecessary; however, the current cast won't affect correctness.